src/backend/distributed/connection/remote_commands.c

```diff
@@ -580,11 +581,18 @@ PutRemoteCopyData(MultiConnection *connection, const char *buffer, int nbytes)
 	 * until the socket is writable to prevent the internal libpq buffers
 	 * from growing excessively.
 	 *
-	 * In the future, we could reduce the frequency of these pushbacks to
-	 * achieve higher throughput.
+	 * We currently allow the internal buffer to grow to 8MB before
+	 * providing back pressure based on experimentation that showed
+	 * throughput get worse at 4MB and lower due to the number of CPU
+	 * cycles spent in networking system calls.
 	 */
 
-	return FinishConnectionIO(connection, allowInterrupts);
+	if (pgConn->outCount > 8 * 1024 * 1024)
```

Comment:
Minor question. Any thoughts on putting the 8MB into a constant variable, or adding parentheses around 8MB?